### PR TITLE
Serialise and deserialise node names, including norad interop

### PR DESCRIPTION
**Pull request by Hoolean**

### CODE REVIEW

File: glyphs_plist/src/font.rs [hunk 1]
```diff
@@ -1121,11 +1142,25 @@ impl NodeType {
...
+                pt.y.into(),
+                node_type.glyphs_str().to_string().into(),
+            ]),
+            Some(attr) => Plist::Array(vec![
+                pt.x.into(),
+                pt.y.into(),
+                node_type.glyphs_str().to_string().into(),
+                attr.to_plist(),
+            ]),
+        }
```
Comment (RickyDaMa) [inline glyphs_plist/src/font.rs]: Saves the duplication perhaps? (Handwritten, not checked to compile or be formatted correctly)

```suggestion
        let mut array = vec![
            pt.x.into(),
            pt.y.into(),
            node_type.glyphs_str().to_string().into(),
        ];
        if let Some(attr) = attr {
            array.push(attr.to_plist());
        }
        Plist::Array(array)
```

File: glyphs_plist/src/norad_interop.rs [hunk 2]
```diff
@@ -22,10 +25,15 @@ impl From<&norad::Contour> for Path {
...
-            path.nodes.iter().map(|node| node.into()).collect();
+impl TryFrom<&Path> for norad::Contour {
+    type Error = norad::error::NamingError;
+
+    fn try_from(path: &Path) -> Result<Self, Self::Error> {
+        let mut points: Vec<norad::ContourPoint> = path
+            .nodes
+            .iter()
+            .map(|node| node.try_into())
+            .collect::<Result<_, _>>()?;
```
Comment (RickyDaMa) [inline glyphs_plist/src/norad_interop.rs]: Nit: I don't like that the type signature is split in two places here

```suggestion
        let mut points = path
            .nodes
            .iter()
            .map(|node| node.try_into())
            .collect::<Result<Vec<norad::ContourPoint>, _>>()?;
```

File: glyphs_plist/src/norad_interop.rs [hunk 3]
```diff
@@ -53,12 +61,18 @@ impl From<&norad::ContourPoint> for Node {
                 (norad::PointType::QCurve, true) => NodeType::QCurveSmooth,
                 (norad::PointType::QCurve, false) => NodeType::QCurve,
             },
+            attr: point.name.as_ref().map(|name| NodeAttrs {
+                name: Some(name.to_string()),
+                other_stuff: Default::default(),
+            }),
```
Comment (RickyDaMa) [inline glyphs_plist/src/norad_interop.rs:67]: Maybe it'd be nice to have a constructor or `From` impl on `NodeAttrs` for doing this (creating `NodeAttrs` from a `Name`)?

One day we'll make it a full-blown builder, but right now that's a very heavy dependency we don't need

File: glyphs_plist/src/norad_interop.rs [hunk 4]
```diff
@@ -308,4 +334,25 @@ mod tests {
...
+            norad::PointType::Move,
+            false,
+            Some(norad::Name::new("hello world").unwrap()),
+            None,
+            None,
+        );
+
+        // Round-trip it.
+        let node = crate::Node::from(&point);
+        let point_again = norad::ContourPoint::try_from(&node).unwrap();
```
Comment (RickyDaMa) [inline glyphs_plist/src/norad_interop.rs:353]: Nit: I'd prefer `expect` here, something like "failed to roundtrip back to norad"

**Review by RickyDaMa — APPROVED** [review]

File: glyphs_plist/src/norad_interop.rs (hunk 3 above)
Comment (Hoolean) [reply glyphs_plist/src/norad_interop.rs:67]: I'll follow the behaviour of the other `Attr` structs for now but let's keep this in our back pocket if we repeat the constructor more than once

File: glyphs_plist/src/norad_interop.rs (hunk 4 above)
Comment (Hoolean) [reply glyphs_plist/src/norad_interop.rs:353]: I left this one in the end as it is in a test

**Pull request merged**